src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
+    private Object[] grow() {
+        return grow(size + 1);
+    }
+
+    private Object[] grow(int minCapacity) {
+        return objects = Arrays.copyOf(objects, newCapacity(minCapacity));
+    }
+
+    private int newCapacity(int minCapacity) {
```

Comment:
1. Why do we need "minCapacity" argument here if we don't use it?
2. It's not good method's name. Method's name should represent some action like "generateNewCapacity" ...

Reply:
done